Signalfx terraform script is crashing when using inside Jenkins

Hi,

Signalfx terraform script is crashing when using inside Jenkins. In Jenkins I'm using the docker image: hashicorp/terraform:full on the slave. And when I run the terraform script in Jenkins I get the following error:

------------------------------------------------------------------------
[31m
[1m[31mError: [0m[0m[1mrpc error: code = Unavailable desc = transport is closing[0m

[0m[0m[0m
panic: runtime error: invalid memory address or nil pointer dereference
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0xde0294]
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: goroutine 12 [running]:
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: github.com/splunk-terraform/terraform-provider-signalfx/signalfx.signalfxConfigure(0xc000325730, 0x0, 0xc0000b0780, 0xc000325730, 0x0)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	github.com/splunk-terraform/terraform-provider-signalfx/signalfx/provider.go:117 +0x874
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Configure(0xc000142900, 0xc00019de60, 0xf31e60, 0xc00019ddd0)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/helper/schema/provider.go:275 +0x102
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).Configure(0xc00009ece0, 0x120e860, 0xc00019daa0, 0xc0002fbd80, 0xc00009ece0, 0xc00019daa0, 0xc000273b78)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/internal/helper/plugin/grpc_provider.go:487 +0x2e7
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_Configure_Handler(0x1040aa0, 0xc00009ece0, 0x120e860, 0xc00019daa0, 0xc0001a2cc0, 0x0, 0x120e860, 0xc00019daa0, 0xc000046600, 0x75)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	github.com/hashicorp/terraform-plugin-sdk@v1.15.0/internal/tfplugin5/tfplugin5.pb.go:3251 +0x214
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000001500, 0x1217d00, 0xc0000a3200, 0xc000183000, 0xc00007e720, 0x17cd0d8, 0x0, 0x0, 0x0)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	google.golang.org/grpc@v1.27.1/server.go:1024 +0x522
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: google.golang.org/grpc.(*Server).handleStream(0xc000001500, 0x1217d00, 0xc0000a3200, 0xc000183000, 0x0)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	google.golang.org/grpc@v1.27.1/server.go:1313 +0xd34
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000038260, 0xc000001500, 0x1217d00, 0xc0000a3200, 0xc000183000)
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	google.golang.org/grpc@v1.27.1/server.go:722 +0xa5
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-10-15T18:38:07.611Z [DEBUG] plugin.terraform-provider-signalfx_v5.0.1: 	google.golang.org/grpc@v1.27.1/server.go:720 +0xa5
2020/10/15 18:38:07 [ERROR] eval: *terraform.EvalConfigProvider, err: rpc error: code = Unavailable desc = transport is closing
2020/10/15 18:38:07 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/10/15 18:38:07 [ERROR] eval: *terraform.EvalOpFilter, err: rpc error: code = Unavailable desc = transport is closing
2020/10/15 18:38:07 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/10/15 18:38:07 [TRACE] [walkPlan] Exiting eval tree: provider["registry.terraform.io/splunk-terraform/signalfx"]
2020/10/15 18:38:07 [TRACE] vertex "provider[\"registry.terraform.io/splunk-terraform/signalfx\"]": visit complete
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.sfx_issuance_journal_1 (expand)" errored, so skipping
2020-10-15T18:38:07.612Z [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.Account_Service_Dashboard_APM_2 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.Account_Service_Dashboard_APM_3 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.Account_Service_Dashboard_APM_5 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.sfx_issuance_journal_3 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.sfx_issuance_journal_0 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_text_chart.sfx_issuance_journal_6 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_text_chart.sfx_issuance_journal_7 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_text_chart.Account_Service_Dashboard_APM_7 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.sfx_issuance_journal_2 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_text_chart.Account_Service_Dashboard_APM_6 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.sfx_issuance_journal_5 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.sfx_issuance_journal_4 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.Account_Service_Dashboard_APM_0 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_dashboard_group.tfgroup (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_single_value_chart.Account_Service_Dashboard_APM_8 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.Account_Service_Dashboard_APM_4 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_time_chart.Account_Service_Dashboard_APM_1 (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_dashboard.Account_Service_Dashboard_APM (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "signalfx_dashboard.sfx_issuance_journal (expand)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "provider[\"registry.terraform.io/splunk-terraform/signalfx\"] (close)" errored, so skipping
2020/10/15 18:38:07 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020/10/15 18:38:07 [INFO] backend/local: plan operation completed
2020-10-15T18:38:07.613Z [DEBUG] plugin: plugin process exited: path=.terraform/plugins/registry.terraform.io/splunk-terraform/signalfx/5.0.1/linux_amd64/terraform-provider-signalfx_v5.0.1 pid=91 error="exit status 2"
2020-10-15T18:38:07.613Z [DEBUG] plugin: plugin exited


Any advice on what this error may be is appreciated.

Thanks

--- a/signalfx/provider.go
+++ b/signalfx/provider.go
@@ -114,10 +114,10 @@ func signalfxConfigure(data *schema.ResourceData) (interface{}, error) {
 	// this additional variable is used for mocking purposes in tests
 	if HomeConfigPath == "" {
 		usr, err := user.Current()
-		HomeConfigPath = usr.HomeDir + HomeConfigSuffix
 		if err != nil {
 			return nil, fmt.Errorf("Failed to get user environment %s", err.Error())
 		}
+		HomeConfigPath = usr.HomeDir + HomeConfigSuffix
 	}
 	if _, err := os.Stat(HomeConfigPath); err == nil {
 		err = readConfigFile(HomeConfigPath, &config)

